src/cpp/src/logit_processor.hpp

```diff
@@ -70,20 +66,19 @@ class TemperatureLogitTransform : public ILogitTransformer {
 public:
     TemperatureLogitTransform(double temperature) : m_temperature(temperature) {};
 
-    std::vector<Token> apply(const std::vector<Token>& input_logits) override {
-        std::vector<Token> output(input_logits.begin(), input_logits.end());
-        std::sort(output.begin(), output.end(), [](const Token& lhs, const Token& rhs) {return lhs.m_log_prob > rhs.m_log_prob; });
-        float max_logit = output[0].m_log_prob;
+    void apply(std::vector<Token>& logits) override {
+        // TODO: No need to sort entire vector if we only need max prob
+        std::sort(logits.begin(), logits.end(), [](const Token& lhs, const Token& rhs) {return lhs.m_log_prob > rhs.m_log_prob; });
```

Comment:
Minor: To replace by `max_element`?

Reply:
Sure, I'll do it, but in the next PR I'll likely propose custom loop here due to performance gains I observed in a draft PR.